Source/BoundaryConditions/ERF_FillPatcher.cpp

```diff
@@ -58,91 +53,106 @@ ERFFillPatcher::ERFFillPatcher (BoxArray const& fba, DistributionMapping  fdm,
-    }
+    if (m_cf_crse_data_old) m_cf_crse_data_old.reset();
+    if (m_cf_crse_data_new) m_cf_crse_data_new.reset();
+    if (m_cf_mask) m_cf_mask.reset();
 
     // Index type for the BL/BA
     IndexType m_ixt = fba.ixType();
 
-    // Box bounding all the fine boxes
-    Box const fine_valid_box = amrex::grow(fba.minimalBox(), IntVect(nghost,nghost,0));
-    Box const fine_valid_box_subset = amrex::grow(fba.minimalBox(), IntVect(nghost_subset,nghost_subset,0));
-
     // Refinement ratios
     for (int idim = 0; idim < AMREX_SPACEDIM; ++idim) {
         m_ratio[idim] = m_fgeom.Domain().length(idim) / m_cgeom.Domain().length(idim);
     }
 
-    // Fine box list (interior halo regions exactly)
-    BoxList bl;
-    bl.set(m_ixt);
-    for (int ibox = 0; ibox < fba.size(); ++ibox) {
-        Box vbx = fba[ibox];
-        BoxList const& bndry = amrex::boxDiff(vbx, fine_valid_box);
-        if (bndry.isNotEmpty()) {
-            bl.join(bndry);
-        }
-    }
-
-    // Fine subset box list (interior halo regions exactly)
-    BoxList bl_subset;
-    bl_subset.set(m_ixt);
-    for (int ibox = 0; ibox < fba.size(); ++ibox) {
-        Box vbx = fba[ibox];
-        BoxList const& bndry = amrex::boxDiff(vbx, fine_valid_box_subset);
-        if (bndry.isNotEmpty()) {
-            bl_subset.join(bndry);
-        }
-    }
-
-    // Coarse box list (coincides with grown fine box list)
+    // Coarse box list
     BoxList cbl;
     cbl.set(m_ixt);
-    cbl.reserve(bl.size());
-    for (auto const& b : bl) {
-        cbl.push_back(interp->CoarseBox(b, m_ratio));
+    cbl.reserve(fba.size());
+    for (int i(0); i<fba.size(); ++i) {
+        cbl.push_back(interp->CoarseBox(fba[i], m_ratio));
     }
 
-    // Box arrays for fine and coarse
-    BoxArray cf_fba(std::move(bl));
+    // Box arrays for the coarse data
     BoxArray cf_cba(std::move(cbl));
-    BoxArray cf_fba_s(std::move(bl_subset));
-    DistributionMapping cf_dm(cf_fba);
-    DistributionMapping cf_dm_s(cf_fba_s);
 
+    // Two coarse patches to hold the data to be interpolated
+    m_cf_crse_data_old = std::make_unique<MultiFab> (cf_cba, fdm, m_ncomp, 0);
+    m_cf_crse_data_new = std::make_unique<MultiFab> (cf_cba, fdm, m_ncomp, 0);
 
-    // Fine patch to hold the time-interpolated state
-    m_cf_fine_data = new MultiFab (cf_fba, cf_dm, m_ncomp, 0);
-
-    // Fine subset patch to hold the time-interpolated state
-    m_cf_fine_subset_data = new MultiFab (cf_fba_s, cf_dm, m_ncomp, 0);
+    // Integer masking array
+    m_cf_mask = std::make_unique<iMultiFab> (fba, fdm, 1, 0);
+    m_cf_mask->setVal(m_relax_mask);
 
-    // Two coarse patches to hold the data to be interpolated
-    m_cf_crse_data[0] = new MultiFab (cf_cba, cf_dm, m_ncomp, 0);
-    m_cf_crse_data[1] = new MultiFab (cf_cba, cf_dm, m_ncomp, 0);
+    // Populate mask array
+    if (nghost_set < 0) {
+        m_cf_mask->setVal(m_set_mask);
+        BuildMask(fba,nghost_set,m_set_mask-1);
+    }
```

Comment:
Line 104 `m_cf_mask->setVal(m_relax_mask);` can be put into the else block of this `if` statement.

Reply:
Addressed in [2fe61a6](https://github.com/erf-model/ERF/pull/1233/commits/2fe61a6ad3ea0c1c1c09868d2f9cf4b13a7908ca)